dao/src/main/java/com/carhouse/dao/CarSaleDao.java

```diff
@@ -35,19 +37,23 @@
      * Add car sale.
      *
      * @param carSale the car sale
+     * @param file    the image file
      * @return car sale id
+     * @throws FileSystemException the file system exception when writing file
      */
-    Integer addCarSale(CarSale carSale);
+    Integer addCarSale(CarSale carSale, MultipartFile file) throws FileSystemException;
```

Comment:
Why it should be placed separately, but not a part of `CarSale` model?

Reply:
As far as I undersend It is not possible to send a file as a field of an object.
Or do you mean to set MultipartFile received from a client to the carSale object in the REST controller?